models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -368,8 +368,10 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
                     ksort($arguments);
                     if (!is_string($transformation['method']) && is_callable($transformation['method'])) {
                         $transformation['method']($image);
+                        $transformation['isApplied'] = true;
                     } elseif (method_exists($image, $transformation['method'])) {
                         call_user_func_array([$image, $transformation['method']], $arguments);
+                        $transformation['isApplied'] = true;
                     }
```

Comment:
```suggestion
                        $transformation['method']($image);
                    } elseif (method_exists($image, $transformation['method'])) {
                        call_user_func_array([$image, $transformation['method']], $arguments);
                    }
                    
                    $transformation['isApplied'] = true;
```

Reply:
Hello @brusch,
just took a quick check on your changes. You r totally right with the 
```php
       if (!empty($transformation) && !$transformation['isApplied'] ?? false) {
```
part.

Regarding the 2nd change:
just wanted to make sure, that you noticed, that the _$transformation['isApplied'] = true;_ was placed intentionally inside the _if_ and _elseif_ part to only set the 'isApplied' value to _true_ if one of the 2 switches applied a transformation. 
If both (if and elseif) statements are false, you would still set _isApplied_ to true.

Reply:
I did some further testings. Seems like PHP 8 still doesn't like
```php
if(!$unsetKey['key'] ?? false) {...}
#but you have to go with the ordinary isset check
```

I'll update my PR and check if the tests run.